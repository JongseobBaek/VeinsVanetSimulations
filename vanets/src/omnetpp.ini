[General]
cmdenv-express-mode = true
cmdenv-autoflush = true
cmdenv-status-frequency = 10000000s

#tkenv-default-config = debug
#tkenv-default-run = 1
tkenv-image-path = bitmaps
ned-path = .

network = RSUScenario

##########################################################
#            Simulation parameters                       #
########################mac##################################
debug-on-errors = true
print-undisposed = false
record-eventlog = true

#6000s
sim-time-limit = 200s

#Extra parameters for testing signals
check-signals=true

#Testing parameters
**.scalar-recording = true
**.vector-recording = true
**.throughputSignal.scalar-recording = true

**.debug = true
**.coreDebug = false

*.playgroundSizeX = 2500m
*.playgroundSizeY = 2500m
*.playgroundSizeZ = 50m

##########################################################
# Annotation parameters                                  #
##########################################################
*.annotations.draw = false

##########################################################
# Obstacle parameters                                    #
##########################################################
*.obstacles.debug = false

##########################################################
#WorldUtilityParameters									 #
##########################################################
*.world.useTorus = false
*.world.use2D = false

##########################################################
#TraCIScenarioManager Parameters						 #
##########################################################
*.manager.updateInterval = 0.1s
*.manager.host = "localhost"
*.manager.port = 9999
*.manager.moduleType = "vanets.src.CarWithMetrics"
*.manager.moduleName = "node"
*.manager.moduleDisplayString = ""
*.manager.autoShutdown = true
*.manager.margin = 25
*.manager.numVehicles = 5
*.manager.launchConfig = xmldoc("erlangen.launchd.xml")

##########################################################
# RSU SETTINGS											 #
##########################################################
*.rsu[0].mobility.x = 2000
*.rsu[0].mobility.y = 2000
*.rsu[0].mobility.z = 3
*.rsu[*].applType = "TraCIDemoRSU11p"
*.rsu[*].appl.debug = false
*.rsu[*].appl.headerLength = 256 bit
*.rsu[*].appl.sendBeacons = false
*.rsu[*].appl.dataOnSch = false
*.rsu[*].appl.sendData = true
*.rsu[*].appl.beaconInterval = 1s
*.rsu[*].appl.beaconPriority = 3
*.rsu[*].appl.dataPriority = 2
*.rsu[*].appl.maxOffset = 0.005s

*.metricsApplication.mobility.x = 2000
*.metricsApplication.mobility.y = 2000
*.metricsApplication.mobility.z = 3
*.metricsApplication.applType = "TraCIDemoRSU11p"
*.metricsApplication.appl.debug = false
*.metricsApplication.appl.headerLength = 256 bit
*.metricsApplication.appl.sendBeacons = false
*.metricsApplication.appl.dataOnSch = false
*.metricsApplication.appl.sendData = true
*.metricsApplication.appl.beaconInterval = 1s
*.metricsApplication.appl.beaconPriority = 3
*.metricsApplication.appl.dataPriority = 2
*.metricsApplication.appl.maxOffset = 0.005s


##########################################################
#            11p specific parameters                     #
#                                                        #
#                    NIC-Settings                        #
##########################################################
*.connectionManager.pMax = 20mW
*.connectionManager.sat = -89dBm
*.connectionManager.alpha = 2.0
*.connectionManager.carrierFrequency = 5.890e9 Hz
*.connectionManager.sendDirect = true

*.**.nic.mac1609_4.useServiceChannel = false
*.**.nic.mac1609_4.scalar-recording = true
 
*.**.nic.mac1609_4.txPower = 20mW
*.**.nic.mac1609_4.bitrate = 6Mbps
*.**.nic.phy80211p.sensitivity = -89dBm
*.**.nic.phy80211p.maxTXPower = 100mW
*.**.nic.mac1609_4.queueSize = ${1,2,5}
*.**.nic.mac1609_4.vector-recording = true

*.**.nic.phy80211p.useThermalNoise = true
*.**.nic.phy80211p.thermalNoise = -110dBm
*.**.nic.phy80211p.decider = xmldoc("config.xml")
*.**.nic.phy80211p.analogueModels = xmldoc("config.xml")
*.**.nic.phy80211p.usePropagationDelay = true

##########################################################
##########################################################
*.node[*].applType = "TraCIDemo11p"
*.node[*].appl.debug = false
*.node[*].appl.headerLength = 256 bit
*.node[*].appl.sendBeacons = false
*.node[*].appl.dataOnSch = false
*.node[*].appl.sendData = true
*.node[*].appl.beaconInterval = 1s
*.node[*].appl.beaconPriority = 3
*.node[*].appl.dataPriority = 2
*.node[*].appl.maxOffset = 0.005s



##########################################################
#                      Mobility                          #
##########################################################
*.node[*].veinsmobilityType = "org.car2x.veins.modules.mobility.traci.TraCIMobility"
*.node[*].mobilityType = "TraCIMobility"
*.node[*].mobilityType.debug = true
*.node[*].veinsmobilityType.debug = true
*.node[*].veinsmobility.x = 0
*.node[*].veinsmobility.y = 0
*.node[*].veinsmobility.z = 1.895
*.node[*0].veinsmobility.accidentCount = 1
*.node[*0].veinsmobility.accidentStart = 10s #75s
*.node[*0].veinsmobility.accidentDuration = 30s

[Config nodebug]
description = "default settings"
**.debug = false
**.coreDebug = false
*.annotations.draw = false

[Config debug]
description = "(very slow!) draw and print additional debug information"
**.debug = true
**.coreDebug = true
*.annotations.draw = true

[Config B_400Bytes_P_1000Bytes_Q_5]
description = "BeaconLength 400Bytes, ServicePacketLength 1000Bytes, QueueSize 5"
**.debug = false
**.coreDebug = false
*.annotations.draw = false
*.rsu[*].appl.beaconLengthBits = 3200bit
*.node[*].appl.beaconLengthBits = 3200bit
*.rsu[*].appl.dataLengthBits = 8000bit
*.node[*].appl.dataLengthBits = 8000bit
*.rsu[*].appl.dataOnSch = true
*.node[*].appl.dataOnSch = true
*.**.nic.mac1609_4.queueSize = 5
*.**.nic.mac1609_4.useServiceChannel = true
*.manager.launchConfig = xmldoc("singlechannel/LowDensityScenario.launchd.xml")

