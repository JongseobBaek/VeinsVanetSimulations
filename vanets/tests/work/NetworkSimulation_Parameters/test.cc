
#include <omnetpp.h>



namespace NetworkSimulation_Parameters {



class Test : public cSimpleModule
{
    public:
        Test() : cSimpleModule(32768) {}
        virtual void activity();
};

Define_Module(Test);

void Test::activity()
{
std::cout<<"WHAT UP " <<std::endl;


}

}; //namespace
