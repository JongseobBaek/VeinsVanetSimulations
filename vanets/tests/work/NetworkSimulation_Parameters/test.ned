package vanets.tests.work.NetworkSimulation_Parameters;

simple Test
{
	@isNetwork(true);
}


