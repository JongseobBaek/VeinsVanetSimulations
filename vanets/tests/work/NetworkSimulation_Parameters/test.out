OMNeT++ Discrete Event Simulation  (C) 1992-2014 Andras Varga, OpenSim Ltd.
Version: 4.6, build: 141202-f785492, edition: Academic Public License -- NOT FOR COMMERCIAL USE
See the license for distribution terms and warranty disclaimer
Setting up Cmdenv...
Loading NED files from .: 2

Preparing for running configuration General, run #0...
Scenario: $repetition=0
Assigned runID=General-0-20160808-14:05:51-8841
Setting up network `Test'...
Initializing...
Initializing module Test, stage 0

Running simulation...
** Event #1  T=0   Test (Test, id=1)
WHAT UP 

<!> No more events -- simulation ended at event #2, t=0.


Calling finish() at end of Run #0...

End.
